src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -45,107 +58,67 @@ export const DropdownMenu: React.FC<Props> = ({
   children,
   className,
   isActive,
-  handleOnKeyDown,
+  handleOnEscDown,
   ...other
 }) => {
-  const childRefs = useRef<Array<HTMLLIElement>>([]);
+  const refs = useRef<Refs>({
+    set: new Set(),
+    list: [],
+  });
+
+  let focusIndex = 0;
+  const focusItem = (index: number) => {
+    refs.current.list[index]
+      ?.querySelector<HTMLButtonElement | HTMLAnchorElement>(':first-child')
+      ?.focus();
+  };
 
   useEffect(() => {
-    setTimeout(() => {
-      if (isActive && childRefs) {
-        childRefs.current[0]
-          ?.querySelector<HTMLButtonElement | HTMLAnchorElement>(':first-child')
-          ?.focus();
-      }
-    }, 1);
+    if (isActive && refs.current.list.length) {
+      focusItem(0);
+    }
   }, [isActive]);
 
-  /**
-   * On KeyDown
-   * 1) Find active item. If there isn't one, do nothing on Keydown
-   * 2) Set active item, next item, and previous item.
-   * 3) If right or down arrow key keyed, focus on next item.
-   * 4) If left or up arrow key keyed, focus on the previous item.
-   */
```

Comment:
Removed comments since the code seemed more self explanatory, lemme know if I should re-add

Reply:
I could handle some inline comments like
```
// Close menu on escape key trigger
// Wrap around to top of menu since we're at the bottom
// Wrap around to bottom of menu since we're at the top
```
or something like that.